### PR TITLE
feat: manually blacklist peers via CLI commands

### PR DESCRIPTION
**Pull request by MicBun**
Review after: https://github.com/trufnetwork/kwil-db/pull/1572

- Add kwild blacklist add/remove/list commands
- Support temporary and permanent blacklisting
- Full CLI integration with Admin RPC backend
- Comprehensive testing guide and documentation

resolves: https://github.com/trufnetwork/kwil-db/issues/1561

See `app/blacklist/TESTING.md` for testing procedure.

## Summary by CodeRabbit

* **New Features**
  * Introduced CLI commands to add, remove, and list blacklisted peers, including support for specifying reasons and durations.
  * Added the ability to manage peer blacklisting independently of private mode.
  * Enhanced admin RPC and JSON-RPC interfaces to support blacklist management with improved error handling and input parsing.
  * Added detailed documentation for testing blacklist CLI commands.

* **Bug Fixes**
  * Fixed issues where blacklist functionality previously required private mode, enabling blacklist-only operation without errors or panics.

* **Tests**
  * Added extensive unit and integration tests for blacklist management, connection gating, and admin service operations, covering edge cases and configuration scenarios.

* **Documentation**
  * Provided a comprehensive testing guide for blacklist CLI usage and validation.

* **Refactor**
  * Improved connection gating logic to support blacklist-only mode and conditional whitelist enforcement.
  * Refactored blacklist manager dependency to use an interface abstraction for better modularity.

### CODE REVIEW

File: app/blacklist/add.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		Duration  string `json:"duration,omitempty"`
+		Permanent bool   `json:"permanent"`
+	}{
+		PeerID:    a.peerID,
+		Reason:    a.reason,
+		Duration:  a.duration,
+		Permanent: a.duration == "",
+	}
+	return json.Marshal(result)
+}
```
[the remark this answers is not in the document]
Comment (MicBun) [reply app/blacklist/add.go]: @coderabbitai Addressed in commit https://github.com/trufnetwork/kwil-db/commit/48b15ad2c543b2701cddf5c2228c614b0542e3e7

File: core/rpc/client/admin/admin.go [hunk 2]
```diff
@@ -30,6 +31,11 @@ type AdminClient interface {
 	RemovePeer(ctx context.Context, peerID string) error
 	ListPeers(ctx context.Context) ([]string, error)
 
+	// Blacklist management
+	BlacklistPeer(ctx context.Context, peerID string, reason string, duration string) error
+	RemoveBlacklistedPeer(ctx context.Context, peerID string) error
+	ListBlacklistedPeers(ctx context.Context) ([]adminjson.BlacklistEntryJSON, error)
+
```
[the remark this answers is not in the document]
Comment (MicBun) [reply core/rpc/client/admin/admin.go]: resolved in https://github.com/trufnetwork/kwil-db/pull/1573/commits/8c8ca867375f4f53a8249250fe0419a6f442d3b1

**Comment by MicBun** [comment]
@MicBun coderabbitai full review

**Comment by MicBun** [comment]
@coderabbitai full review

**Comment by MicBun** [comment]
Cannot merge, can this be approved again? @outerlook
